tidy3d/components/data/monitor_data.py

```diff
@@ -553,7 +566,15 @@ def outer_dot(
                     dot[i, mi0, mi1] = 0.25 * integrand.sum(dim=d_area.dims)
 
         coords = {"f": f, "mode_index_0": mode_index_0, "mode_index_1": mode_index_1}
-        return MixedModeDataArray(dot, coords=coords)
+        result = MixedModeDataArray(dot, coords=coords)
+
+        # Remove mode index coordinate if the input did not have it
+        if not modes_in_self:
+            result = result.isel(mode_index_0=0, drop=True)
+        if not modes_in_other:
+            result = result.isel(mode_index_1=0, drop=True)
+
+        return result
```

Comment:
This is a bit weird in that the result will be a `MixedModeDataArray` if both `self` and `other` are `ModeFieldData`, but `xr.DataArray` otherwise. I think there is no issue with just always returning an `xr.DataArray`? We do have some data methods that return xarray data: for example, `colocate`. More generally it's easy for a user to move from our custom data types to xarray, e.g. as soon as they do `sel` or `isel`. @tylerflex remind me if there was any specific logic to when we should use our custom data types and if you think it's fine to just return `xr.DataArray` here (so just remove MixedModeDataArray's definition entirely, as it is only used here).

Reply:
I think if it's something stored internally, we should use our own custom data types, for the result of a method or property, using xr.DataArray or xr.Dataset is fine. and maybe preferable

Reply:
Sounds good, @lucas-flexcompute could you make this change? I guess it's just making the result an `xr.DataArray` from the start, and you don't need to remove the `MixedModeDataArray` even though it's only used here (maybe we'll use it in the future). I was wondering if this can also help simplify this method (a lot of `if` checks for a mode dimension currently), but I don't see an obvious way.

Reply:
I though about it as well and the alternatives I could think of looked more cumbersome then the current ifs.